File: popfragg.Domain/DTOS/Steam/SignUpSteamRequest.cs
using popfragg.Domain.DTOS.Authorizer.Requests;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace popfragg.Domain.DTOS.Steam
{
    public class SignUpSteamRequest
    {
        public string Email { get; set; } = default!;
        public string Password { get; set; } = default!;
        public string ConfirmPassword { get; set; } = default!;
        public string[] Scope { get; set; } = default!;
        public string Nickname { get; set; } = default!;
        public string? GivenName { get; set; }
        public string? FamilyName { get; set; }
        public string? Gender { get; set; }
        public string? Birthdate { get; set; }
        public string? PhoneNumber { get; set; }
        public AppDataRequest AppData { get; set; } = default!;
    }
}

File: popfragg.Tools.Codegen/Outputs/schemas/sign-up-steam-request.ts
import { z } from 'zod';

export const signUpSteamSchema = z.object({
  email: z.string().optional(),
  password: z.string().optional(),
  confirmPassword: z.string().optional(),
  scope: z.any().optional(),
  nickname: z.string().optional(),
  givenName: z.string().optional(),
  familyName: z.string().optional(),
  gender: z.string().optional(),
  birthdate: z.string().optional(),
  phoneNumber: z.string().optional(),
  appData: appDataRequestSchema.optional(),
});

export type SignUpSteamRequest = z.infer<typeof signUpSteamSchema>;
